resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/K8sSubmitOps.scala

```diff
@@ -0,0 +1,180 @@
+  }
+
+  override def executeOnGlob(pods: List[Pod], ns: String, gracePeriodSeconds: Option[Long])
+      (implicit client: KubernetesClient): Unit = {
+    if (pods.nonEmpty) {
+      pods.foreach(pod => printMessage(s"Deleting driver pod: ${pod.getMetadata.getName}."))
+      val nClient = client
+        .pods
+        .inNamespace(ns)
+      gracePeriodSeconds match {
+        case Some(period) =>
+          // this is not using the batch api because no option is provided
+          // when using the grace period.
+          pods.foreach { pod =>
+             nClient
+              .withName(pod.getMetadata.getName)
+               .withGracePeriod(period)
+              .delete()
+          }
+        case _ => nClient.delete(pods.asJava)
+      }
+    } else {
+      printMessage("No applications found.")
+    }
+  }
+}
+
+private [spark] class ListStatus extends K8sSubmitOp {
+  override def executeOnPod(pName: String, ns: String, gracePerio: Option[Long])
+      (implicit client: KubernetesClient): Unit = {
+      val pod = client
+        .pods
+        .inNamespace(ns)
+        .withName(pName)
+        .get()
+
+      printMessage("Application status (driver): " +
+        Option(pod).map(formatPodState).getOrElse("unknown."))
+  }
+
+  override def executeOnGlob(pods: List[Pod], ns: String, gracePeriod: Option[Long])
+      (implicit client: KubernetesClient): Unit = {
+    if (pods.nonEmpty) {
+      for (pod <- pods) {
+        printMessage("Application status (driver): " +
+          Option(pod).map(formatPodState).getOrElse("unknown."))
+      }
+    } else {
+      printMessage("No applications found.")
+    }
+  }
+}
+
+private[spark] class K8sSparkSubmitOperationsClient extends SparkSubmitOperationsClient
+  with CommandLineLoggingUtils {
+  private def isGlob(name: String): Boolean = {
+    name.last == '*'
+  }
+
+  def execute(submissionId: String, sparkConf: SparkConf, op: K8sSubmitOp): Unit = {
+    val master = KubernetesUtils.parseMasterUrl(sparkConf.get("spark.master"))
+    val sIdParts = submissionId.split(":") match {
+      case Array(ns, pName) => Some((ns, pName))
+      case Array(pName) => Some(("default", pName))
+      case _ => None
+    }
+
+    sIdParts match {
+      case Some((ns, pName)) =>
+        Utils.tryWithResource(SparkKubernetesClientFactory.createKubernetesClient(
+          master,
+          Some(ns),
+          KUBERNETES_AUTH_SUBMISSION_CONF_PREFIX,
+          sparkConf,
+          None,
+          None)) { kubernetesClient =>
+          implicit val client: KubernetesClient = kubernetesClient
+          if (isGlob(pName)) {
+            val pods = {
+              kubernetesClient
+                .pods
+                .inNamespace(ns)
+                .list()
+                .getItems
+                .asScala
+                .filter { pod =>
+                  val meta = pod.getMetadata
+                  meta.getName.startsWith(pName.stripSuffix("*")) &&
+                    meta
```

Comment:
isn't this the same as `meta.getLabels().get(SPARK_ROLE_LABEL) == SPARK_POD_DRIVER_ROLE`?

Reply:
Ping.